src/FlaUI.WebDriver/Controllers/ActionsController.cs

```diff
@@ -23,18 +21,30 @@ public ActionsController(ILogger<ActionsController> logger, ISessionRepository s
         public async Task<ActionResult> PerformActions([FromRoute] string sessionId, [FromBody] ActionsRequest actionsRequest)
         {
             var session = GetSession(sessionId);
-            var actionsByTick = ExtractActionSequence(actionsRequest);
+            var actionsByTick = ExtractActionSequence(session, actionsRequest);
             foreach (var tickActions in actionsByTick)
             {
                 var tickDuration = tickActions.Max(tickAction => tickAction.Duration) ?? 0;
-                var dispatchTickActionTasks = tickActions.Select(tickAction => DispatchAction(session, tickAction));
+                var dispatchTickActionTasks = tickActions.Select(tickAction => ActionsDispatcher.DispatchAction(session, tickAction));
                 if (tickDuration > 0)
                 {
                     dispatchTickActionTasks = dispatchTickActionTasks.Concat(new[] { Task.Delay(tickDuration) });
                 }
                 await Task.WhenAll(dispatchTickActionTasks);
             }
 
+            // The spec says that input sources must be created for actions, but then doesn't specify how or when they should be
```

Comment:
I think this step isn't needed because Selenium will itself call the ReleaseActions endpoint. That will dispatch the release actions and clear the input state, which removes all input sources.

Reply:
I've not observed `ReleaseActions` being called automatically by Selenium. For example if I run the `PerformActions_KeyDownKeyUp_IsSupported` test and put a breakpoint on `ActionsController.ReleaseActions` it doesn't get hit.

Reply:
Ok I guess I should remove this code then. One thing I've noticed though: the [Avalonia integration tests](https://github.com/AvaloniaUI/Avalonia/tree/master/tests/Avalonia.IntegrationTests.Appium) use a long-running session and by the end of that session, things get _really_ slow. I need to investigate but I wonder if the input state not being cleared has something to do with it. Obviously I can add a `ResetInputState` at the beginning of each test, but this might be a complaint (if this is indeed the cause of the slowdown).

Reply:
Code removed!